tests/openapi/test_service.py

```diff
@@ -58,8 +60,42 @@ def test_telemetry_detailed():
 
     endpoint = result['requests']['rest']['responses']['PUT /collections/{name}/points']
     assert endpoint['200']['count'] > 0
+    
+    if level == 0:
+        'collections' not in result['collections']
+        return
 
     last_queried = endpoint['200']['last_responded']
     last_queried = datetime.fromisoformat(last_queried)
     # Assert today
     assert last_queried.date() == datetime.now().date()
+
+    collection = result['collections']['collections'][0]
+
+    if level == 1:
+        assert list(collection.keys()) == ['vectors', 'optimizers_status', 'params']
+    elif level == 2:
+        assert list(collection.keys()) == ['id', 'init_time_ms', 'config']
+    elif level >= 3:
+        assert list(collection.keys()) == ['id', 'init_time_ms', 'config', 'shards', 'transfers', 'resharding']
+
+    if level >= 3:
+        shard = collection['shards'][0]
+        assert list(shard.keys()) == ['id', 'key', 'local', 'remote', 'replicate_states']
+
+        local_shard = shard['local']
+
+        if level == 3:
+            assert list(local_shard.keys()) == [
+                'variant_name', 'status', 'total_optimized_points', 'vectors_size_bytes',
+                'payloads_size_bytes', 'num_points', 'num_vectors', 'optimizations', 'async_scorer'
+            ]
+        elif level > 3:
+            assert list(local_shard.keys()) == [
+                'variant_name', 'status', 'total_optimized_points', 'vectors_size_bytes',
+                'payloads_size_bytes', 'num_points', 'num_vectors', 'segments', 'optimizations', 'async_scorer'
+            ]
+
+    if level >= 4:
+        segment = local_shard['segments'][0]
+        assert list(segment.keys()) == ['info', 'config', 'vector_index_searches', 'payload_field_indices']
```

Comment:
I am not sure we care that much about those levels. It makes this test kind of fragile. Also, `.keys()` ordering is not guaranteed

Reply:
> Also, .keys() ordering is not guaranteed

It returns ordered_dict and Qdrant always returns them in same order. it's not gonna be flaky.

> I am not sure we care that much about those levels

yeah it's not critical piece of code. Better to have tests than not have them?

Reply:
> It returns ordered_dict and Qdrant always returns them in same order. 

nothing really enforces that

Reply:
Using `set` to avoid order change issues now.